subsys/bluetooth/host/hci_core.c

```diff
@@ -3724,6 +3724,9 @@ int bt_disable(void)
 	/* Some functions rely on checking this bitfield */
 	memset(bt_dev.supported_commands, 0x00, sizeof(bt_dev.supported_commands));
 
+	/* If random address was set up - clear it */
+	bt_addr_le_copy(&bt_dev.random_addr, BT_ADDR_LE_ANY);
+
```

Comment:
Given that `bt_dev` is defined as

```
/* State tracking for the local Bluetooth controller */
```

It sort of feels like we should be able to do a `memset(&bt_dev, 0, sizeof(bt_dev))`, but I do see that we use it for a lot more than what it is defined as. 

I just can't help but to wonder if this solution here is just plugging one hole, and more holes may appear/haven't been found yet if we do not find a more longterm solution (e.g. splitting `bt_dev` into multiples, and one of them being the BT controller state that we can just memset on `bt_disable` (or on `bt_enable` when we send the `hci reset` command).

Reply:
https://github.com/zephyrproject-rtos/zephyr/issues/49215